### PR TITLE
2.x branch - Publish snapshots to maven via GHA

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Publish snapshots to maven via GHA.
Cherry-picked the changes in the `main` branch PR https://github.com/opensearch-project/common-utils/pull/362
 
### Issues Resolved
https://github.com/opensearch-project/common-utils/issues/361

### Tests
Tested locally on my `2.x`-based branch.
```
$ find snapshots | sort   
snapshots
snapshots/org
snapshots/org/opensearch
snapshots/org/opensearch/common-utils
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-javadoc.jar
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-javadoc.jar.md5
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-javadoc.jar.sha1
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-javadoc.jar.sha256
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-javadoc.jar.sha512
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-sources.jar
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-sources.jar.md5
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-sources.jar.sha1
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-sources.jar.sha256
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1-sources.jar.sha512
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.jar
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.jar.md5
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.jar.sha1
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.jar.sha256
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.jar.sha512
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.pom
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.pom.md5
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.pom.sha1
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.pom.sha256
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/common-utils-2.6.0.0-20230222.015350-1.pom.sha512
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/maven-metadata.xml
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/maven-metadata.xml.md5
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/maven-metadata.xml.sha1
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/maven-metadata.xml.sha256
snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/maven-metadata.xml.sha512
snapshots/org/opensearch/common-utils/maven-metadata.xml
snapshots/org/opensearch/common-utils/maven-metadata.xml.md5
snapshots/org/opensearch/common-utils/maven-metadata.xml.sha1
snapshots/org/opensearch/common-utils/maven-metadata.xml.sha256
snapshots/org/opensearch/common-utils/maven-metadata.xml.sha512

```
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by sbcd90 — APPROVED** [review]

**Comment by gaiksaya** [comment]
Hey @AWSHurneyt ,

Posting the same query here. The current snapshots do not include javadoc and sources jar 
https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/common-utils/2.6.0.0-SNAPSHOT/
Can you check why the above is doing the same? Maybe gradle task is the issue?

**Comment by gaiksaya** [comment]
Okay this is the culprit https://github.com/opensearch-project/common-utils/pull/365/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R183-R184

**Comment by gaiksaya** [comment]
I saw this https://github.com/opensearch-project/common-utils/blob/main/scripts/build.sh#L68 maybe what what we want to do?

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+      - name: publish snapshots to maven
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
+          ./gradlew publishAllPublicationsToSnapshotsRepository
```
Comment (gaiksaya) [inline .github/workflows/maven-publish.yml:39]: Please change the task to `publishShadowPublicationToSnapshotsRepository` that was being used previously

Comment (AWSHurneyt) [reply .github/workflows/maven-publish.yml]: Will do.

**Review by gaiksaya — APPROVED** [review]

**Comment by gaiksaya** [comment]
Added the secret

**Review by Angie-Zhang — APPROVED** [review]

**Pull request merged**